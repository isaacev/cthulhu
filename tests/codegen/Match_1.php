<?php

namespace Match_1 {
  function main() {
    $b = 2 + 2;
    if ($b == 0) {
      $c = print("zero" . "\n");
    } else if ($b == 1) {
      $c = print("one" . "\n");
    } else if ($b == 2) {
      $c = print("two" . "\n");
    } else if (true) {
      $c = print("several" . "\n");
    } else {
      die("match expression did not cover all possibilities\n");
    }
    $c;
  }
}

namespace {
  \Match_1\main(null);
}
