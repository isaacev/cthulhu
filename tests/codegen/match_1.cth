#[entry]
fn main() -> () {
  match 2 + 2 {
    0 => ::Io::println("zero"),
    1 => ::Io::println("one"),
    2 => ::Io::println("two"),
    _ => ::Io::println("several"),
  };
}
