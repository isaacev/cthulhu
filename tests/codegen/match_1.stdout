<?php

namespace Kernel\Builtins {
  // #[inline]
  // #[construct]
  function _print($a) {
    print($a);
  }
}

namespace Io {
  // #[inline]
  function println($str) {
    \Kernel\Builtins\_print($str . "\n");
  }
}

namespace match_1 {
  // #[entry]
  function main() {
    $b = 2 + 2;
    if ($b == 0) {
      \Io\println("zero");
    } else if ($b == 1) {
      \Io\println("one");
    } else if ($b == 2) {
      \Io\println("two");
    } else if (true) {
      \Io\println("several");
    }
    $a;
  }
}

namespace {
  \match_1\main();
}
